go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -217,14 +218,20 @@ func (p DoltDatabaseProvider) databaseForRevision(ctx context.Context, revDB str
 	if !ok {
 		return nil, dsess.InitialDbState{}, false, nil
 	}
-	srcDb, ok := candidate.(Database)
+
+	srcDb, ok := candidate.(SqlDatabase)
 	if !ok {
 		return nil, dsess.InitialDbState{}, false, nil
 	}
 
-	if isBranch(ctx, srcDb.ddb, revSpec) {
-		// if the requested revision is a br we can
-		// write to it, otherwise make read-only
+	if replicaDb, ok := srcDb.(ReadReplicaDatabase); ok {
+		err := switchAndFetchReplicaHead(ctx, revSpec, replicaDb)
```

Comment:
I think this is fine for now, but it's not going to work longer term.

The issue is that this method is called primarily during analysis, *before* any IO should in principle happen. We want the fetch and pull to happen when the query begins to execute, not here.

Would add a TODO for now